core/controllers/profile_test.py

```diff
@@ -244,6 +244,23 @@ def test_get_profile_picture_valid_username(self):
 
 class ProfileDataHandlerTests(test_utils.GenericTestBase):
 
+    def test_preference_page_updates(self):
+        self.signup(self.EDITOR_EMAIL, username=self.EDITOR_USERNAME)
+        self.login(self.EDITOR_EMAIL)
+        response = self.testapp.get('/preferences')
+        csrf_token = self.get_csrf_token_from_response(response)
+        self.put_json(
+            '/preferenceshandler/data',
+            {'update_type': 'preferred_site_language_code', 'data': 'en'},
+            csrf_token=csrf_token)
+        self.put_json(
+            '/preferenceshandler/data',
+            {'update_type': 'preferred_language_codes', 'data': ['en']},
+            csrf_token=csrf_token)
+        response = self.get_json('/preferenceshandler/data')
+        self.assertEqual(response['preferred_language_codes'], ['en'])
```

Comment:
To make this robust, you'd probably want to test that, before the puts, that both entries are _not_ equal to 'en' and ['en'] respectively.